src/selectors/modular-pipelines.js

```diff
@@ -44,7 +44,12 @@ export const searchTree = (
       // if the child node is a leaf, simply search the leaf's name
       // and add to the search result if there is a match.
       const found = searchString(childNode.data.name, searchValue);
-      if (found) {
+      const foundOpposite = searchString(
+        childNode.data.oppositeForPrettyName,
+        searchValue
+      );
+
+      if (found || foundOpposite) {
```

Comment:
This works in searching for both pretty and non-pretty names but the node list displayed is changed.

Have a look at the gif - The display names in the list gets changed based on search value i.e., I disabled pretty name flag and searched for a pretty name "Feat" and the node list shows Feature Engineering instead of feature_engineering. Though the search results return a value, we should push names based on the pretty flag I guess

![Display_Name changes](https://github.com/kedro-org/kedro-viz/assets/87735534/ae14661f-38a3-4dde-bc63-dc9101d94904)

Reply:
I checked current version on https://demo.kedro.org and this is happening before my changes. We can create a separate ticket for this if this behaviour is not as expected.

Reply:
No, that isn't expected. @jitu5 is right though, this isn't due to his change.

In the screen capture, what's returned when you search for `feat` when `prettyName` is **off** should be `feature_engineering` and not `Feature Engineering`. 

Jitendra, can you please create a separate ticket to fix that?